### ISSUE TITLE
Error when parsing empty list with subcast

### ISSUE THREAD

**Issue by sabdouni** [sloria/environs#137]
Hello,

The `env.list` can't handle an empty list when using `subcast`
It raises an exception `environs.EnvValidationError: Environment variable "LIST" invalid: {0: ['Not a valid integer.']}`
This can be reproduced by e.g 
```
    os.environ["LIST"] = ""
    env = Env()
    LIST = env.list("LIST", [], subcast=int)
```
If you don't use `subcast` it works just fine by e.g 
```
    os.environ["LIST"] = ""
    env = Env()
    LIST = env.list("LIST", [])
```

Maybe one solution is if the env variable is falsy (empty or doesn't exist) `env.list` should return the default value if defined

Happy to submit a PR if that would be of interest. Thanks!

**Comment by sloria** [comment]
Thanks for reporting. PRs welcome! =)

**Comment by sabdouni** [comment]
@sloria the bug is located in the method `_preprocess_list` and is not related really to environs code but to a really weird behavior of 'split',

**Comment by sloria** [comment]
Seems like a reasonable fix

**Comment by sloria** [comment]
Fix is released in 7.3.1
